packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -114,22 +115,36 @@ class SocketReference extends TypedEventEmitter<ISocketEvents> {
         assert(!SocketReference.socketIoSockets.has(key), 0x220 /* "socket key collision" */);
         SocketReference.socketIoSockets.set(key, this);
 
-        // The server always closes the socket after sending this message
-        // fully remove the socket reference now
-        socket.on("server_disconnect", (socketError: IOdspSocketError) => {
+        const disconnectHandler = (socketError: IOdspSocketError, clientId?: string) => {
             // Treat all errors as recoverable, and rely on joinSession / reconnection flow to
             // filter out retryable vs. non-retryable cases.
-            const error = errorObjectFromSocketError(socketError, "server_disconnect");
+            const error = errorObjectFromSocketError(socketError, clientId
+                ? "disconnect_document" : "server_disconnect");
```

Comment:
I'd rather have code on lines 142 & 147 pass in `handle: string` argument and this code not care about these differences.